app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js

```diff
@@ -138,6 +138,38 @@ discourseModule("Unit | Utility | uploads", function () {
     );
   });
 
+  test("shows error message when no extensions are authorized", function (assert) {
+    this.siteSettings.authorized_extensions = "";
+    this.siteSettings.authorized_extensions_for_staff = "";
+
+    sinon.stub(dialog, "alert");
+    assert.notOk(
+      validateUploadedFiles([{ name: "test.jpg" }], {
+        user: User.create(),
+        siteSettings: this.siteSettings,
+      })
+    );
+    assert.ok(
+      dialog.alert.calledWith(I18n.t("post.errors.no_uploads_authorized"))
+    );
+  });
+
+  test("shows error message when no extensions are authorized for staff", function (assert) {
+    this.siteSettings.authorized_extensions = "";
+    this.siteSettings.authorized_extensions_for_staff = "";
```

Comment:
```suggestion
    this.siteSettings.authorized_extensions_for_staff = "";
```

I think we only need to change one site setting for this test case.

Reply:
Both site settings are taken into account because the test user is staff.